nio/client/async_client.py

```diff
@@ -166,14 +170,24 @@ class AsyncClientConfig(ClientConfig):
             For example, with the default backoff_factor of 0.1,
             nio will sleep for 0.0, 0.2, 0.4, ... seconds between retries.
 
-        max_timeout_retry_wait_time (float): The maximum time to wait between
-            retries for timeouts, by default 60.
+        max_timeout_retry_wait_time (float): The maximum time in seconds to
+            wait between retries for timeouts, by default 60.
+
+        timeout (float): How many seconds a request has to finish, before it is
```

Comment:
I think it would be better to name this `request_timeout` since we already have a sync timeout.